data/src/main/kotlin/com/antosik/benchproject/data/popular/movies/repo/PopularMoviesDataRepository.kt

```diff
@@ -14,9 +17,30 @@ internal class PopularMoviesDataRepository(
 
     override suspend fun getPopularMoviesList(): List<MovieModel> {
         runCatching {
-            val remoteMovies = apiService.getPopularMovies(BuildConfig.API_KEY)
-            dao.insertMovies(remoteMovies.toEntity())
+            dao.insertMovies(fetchAndMergeMoviesWithLocalData())
         }
         return dao.getMovies().toDomain()
     }
+
+    override suspend fun updatePopularMovie(movieModel: MovieModel) {
+        runCatching {
+            dao.updateMovie(movieModel.toEntity())
+        }
+    }
+
+    private suspend fun fetchAndMergeMoviesWithLocalData(): List<MovieEntity> {
+        val entityMovies = dao.getMovies()
+        val remoteMovies = apiService.getPopularMovies(BuildConfig.API_KEY)
+        return remoteMovies.results.map { remoteMovie ->
+            val entityMovie = entityMovies.find { it.movieId == remoteMovie.id }
+            MovieEntity(
```

Comment:
shouldn't this be done in a separate mapper ?

Reply:
Yes, of course, I can do this, the mapper will be a little bit complicated because of the parameter for `isFavorite`

Reply:
Done 😄